crates/core/src/octree/build/stages/allocate_nodes.rs

```diff
@@ -57,3 +63,157 @@ impl AllocateNodesPass {
         self.shader.wait();
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use std::path::PathBuf;
+    use std::env;
+
+    struct TestCase {
+        pub input: AllocateNodesInput,
+        pub expected_output: Vec<u32>,
+        pub description: String,
+    }
+
+    #[test]
+    fn allocate_nodes_works() {
+        let (_glfw, _window) = test_utils::init_opengl_context();
+
+        // To go from the crate root to the workspace root
+        // This is stateful, so not a good testing practice, doing it on first test ran for now
+        let mut path = PathBuf::from(env::current_dir().unwrap());
+        path.pop();
+        path.pop();
+        env::set_current_dir(path).unwrap();
+
+        let voxel_dimension_exponent = 4;
+
+        unsafe {
+            let test_data = test_cases();
+
+            // Initialize everything
+            Config::initialize_test_sensitive(Config::new(voxel_dimension_exponent), true);
+            let allocate_nodes_pass = AllocateNodesPass::init();
+            let mut allocations: Vec<u32> = vec!();
```

Comment:
```suggestion
            let mut allocations: Vec<u32> = vec![];
```

I've just never seen it with parenthesis before 😅, I know it works